ckan/controllers/user.py

```diff
@@ -162,10 +166,90 @@ def register(self, data=None, errors=None, error_summary=None):
 
         return self.new(data, errors, error_summary)
 
+    def _generate_internal_logs(self, log_message):
+        '''
+        This helper function is called if sending a new user registration email fails.
+        '''
+
+        logging.warning("There was an error sending the email. Writing to log file.")
+
+        os.chdir(os.environ['REGISTRATION_ERROR_LOG_FILE_PATH'])
+
+        with open(os.environ['REGISTRATION_ERROR_LOG_FILE_NAME'], "a") as fp:
+            try:
+                fp.write(log_message)
+                fp.write("\n\n")
+            except:
+                logging.warning("There is an error writing to the log file. Dumping the output to console.")
+                logging.warning(log_message)
+
+        logging.warning("Written to log file.")
+
+    def email_new_user_details_using_mailgun(self, request_data):
+        request_params = dict(request_data)
+
+        email_body = "There is a new user registration request. \
+        \nThe user's details are as follows: \
+        \n\
+        \nUsername: {0}\
+        \nName: {1} \
+        \nEmail: {2} \
+        \nReason for Request: {3} \
+        \nProject of Interest: {4} ".format(request_params['name'], request_params['fullname'], request_params['email'], request_params['request_reason'], request_params['project_of_interest'])
+
+        MAILGUN_VARS = {
```

Comment:
style: I wouldn't bother making this temporary placeholder dict, it doesn't buy you anything. just use a generator expressinon and do something like `MAILGUN_VARS = dict((t, os.envorn.get(t)) for t in [list of environment variables]))` and then check if there are any `None` in the values (you can use `any` to do that)

Reply:
(and then you can drop all those flag variables like unset_var, which are usually a bad sign stylistically)

Reply:
Fixed. 

I used a for loop (3 lines) to check for None instead of `any` because `any` seems to return only True/False? If there was a missing variable I was wanting to print out the key. I did a bit of Googling to see if I could use `any` to return the values which were triggering the True value but didn't find anything too straightforward. Is what I did OK?